Library/Homebrew/extend/pathname.rb

```diff
@@ -189,6 +189,7 @@ def atomic_write(content)
       rescue Errno::EPERM # rubocop:disable Lint/HandleExceptions
       end
 
+      tf.close # Close the file before renaming to prevent 'Device or resource busy' error
       File.rename(tf.path, self)
     ensure
       tf.close!
```

Comment:
Is there no error if this is closed twice?

Reply:
Closing the file twice does not produce an error.